alumnos/mpp/ejercicios-clase-3.js

```diff
@@ -0,0 +1,105 @@
+// 2PIr
+function perimetroCirculo(r) {
+    return 2 * Math.PI * r;
+}
+
+console.assert(perimetroCirculo(3) === 18.84955592153876)
+console.assert(perimetroCirculo(6) === 37.69911184307752)
+
+function areaCirculo(r) {
+    // πr^2
+    return Math.PI * r ** 2;
+}
+
+console.assert(areaCirculo(2) === 12.566370614359172)
+console.assert(areaCirculo(8) === 201.06192982974676)
+
+// h^2 = c1^2 + c2^2
+function hipotenusa(c1, c2) {
+    return Math.sqrt(c1 ** 2 + c2 ** 2);
+}
+
+console.assert(hipotenusa(1, 2) === 2.23606797749979)
+
+function azarMoneda() {
+    return (Math.random() >= 0.5) ? "Cara" : "Cruz";
+}
+
+azarMoneda();
+
+function dado() {
+    return Math.floor(Math.random() * 6 + 1);
+}
+
+function azarDados() {
+    let resultado = 0;
+    for (let i = 0; i < 100; ++i) {
+        if (dado() + dado() === 10) {
+            resultado++;
+        }
+    }
+    return resultado;
+}
+
+azarDados();
+
+function repite(str, n) {
+    let resultado = '';
+    for (let i = n; i > 0; --i) {
+        resultado += str;
+    }
+    return resultado;
+}
+
+console.assert(repite('Batman ', 3) === 'Batman Batman Batman ')
+
+function palabraMasLarga(str) {
+    let masLarga = '';
+    for (let palabra of str.split(' ')) {
+        if (palabra.length > masLarga.length) {
+            masLarga = palabra;
+        }
+    }
+    return masLarga;
+}
+
+console.assert(palabraMasLarga("Erase una vez que se era") === "Erase")
+
+function ponPrimeraMayuscula(str) {
+    let nuevoStr = '';
+    for (let palabra of str.split(' ')) {
```

Comment:
El ejercicio está perfecto 😄 

Esto es sólo un detalle, cuando en un for pones que sea en base a la ejecución de algo (`str.split(' ')` en este caso), dicho código se ejecuta en cada iteración. Para pocas iteraciones sin problema, pero si tienes que iterar por un listado con muchos elementos, puede afectar al rendimiento de tu programa.

Reply:
No lo había pensado. Mejor guardar `str.split(' ')` en una variable antes de empezar el bucle. Muchas gracias!